### DISCUSSION TITLE
Linking libz requirment on AIX

### DISCUSSION THREAD

**Discussion by timlegge** [openssl/openssl#24628] [Q&A]
Hi

I was asked to look at an issue in an issue with an OpenSSL based perl module on AIX and found that a number of OpenSSL based modules required the explicit linking of libz.

Granted, it is an old version of openSSL (OpenSSL 1.0.2u  20 Dec 2019) so I (and they) know it is unsupported.  My question is:

1. Are there specific versions of openssl that would require me to add -lz to the linking options or
2. Is this likely AIX related?

I have not seen this issue on any other platform so I am leaning towards it being an AIX specific thing.

thanks

**Comment by nhorman** [comment]
If you enabled zlib in openssl, then yes, libz is required to be linked, but the build script would have done that for you.

but it sounds like you're building some perl module that uses openssl.  The details for that are going to be specific to that perl module

**Reply by timlegge** [reply]
Hi

I found it necessary for:

Crypt::OpenSSL::Random
Crypt::OpenSSL::RSA
Crypt::OpenSSL::PKCS10
Crypt::OpenSSL::PKCS12

Not asking you guys to review :-) .

The modules work fine on Linux etc. without the option specified in the Makefile.PL so I assume it may be dynamically loading libz there and not doing it on AIX...

Tim
